rfcs/packaging/package-structure.md

```diff
@@ -0,0 +1,134 @@
+# RFC: Communicating new package structure for `@fluentui/*`
+
+**Note that this RFC focuses solely on the package structure from a consumer perspective, focused primarily on v7/8 and converged packages. It does not deal with folder structure or northstar packages.**
+
+Contributors: @paulgildea, @JustSlone, @ecraig12345, @khmakoto
+
+## Summary
+
+Currently, we have a lot of different component packages, and it's not clear to consumers what the purpose of each one is or whether it's production-ready.
+
+There are a few aspects to the proposed solution:
+
+- Add badges (or similar) to package readmes to clarify GA/preview/dev status
+- "Converged" component packages only contain convergence work
+  - Move old (v8) component implementations from `react-{checkbox,link,slider,tabs,toggle}` back to `react-internal`
+  - Move prototype "converged" components to the package roots and get rid of `/next` folders
+  - Probably also move all component implementations in `react` back to `react-internal`
+    - OR (open question) just get rid of `react-internal`?
+- _(after v8 release)_ Add `@fluentui/react-preview` package which exports _only_ converged components
```

Comment:
This package will live in the same branch as the current version of `@fluentui/react`. As far as we've figured out there's not a good way to have a package of the same name be published out of the same branch with different tags.

Reply:
The new package also won't be a 1-1 equivalent if we stick with the plan that it should only export preview components, so having it under the same name might be misleading.

Reply:
called me old fashioned, but I'd suggest to use some visualizations, from my experience it makes things more obvious/understandable. 

Do I understand it correctly that we wanna achieve something like following ?

**Plan**
![image](https://user-images.githubusercontent.com/1223799/105991800-89ef5a80-60a4-11eb-8103-a499416a4e7b.png)

**Future**
![image](https://user-images.githubusercontent.com/1223799/105992250-2ade1580-60a5-11eb-9d06-e59e47c03d7b.png)

Reply:
I agree on the need for a suite package for properly converge components. Let's not call it preview as it will be the production versions and we'll likely indefinitely get stuck with the "preview" label. Ideally, we could call it `@fluentui/react` since that is the end goal. If that is not possible due to v7/8 using that namespace, then I see two more options:

### v9

Since 1) development on 7/8 and 0 are essentially in maintenance/bug fix mode only 2) work will only be done on convergence 3) react-* packages will be used for converged components only, THEN why not also use the `@fluentui/react` package name for converged only and just make it v9? We could make branches for 7/8 and 0 where those will be maintained and moving forward just have v9 on master.

### Pick a new name

Use something other than "-preview" like "-components" or other non-colliding name. The name should not be point in time sensitive, though.

Reply:
Will the v9 approach work for mix and match between v8 components and converged components? I think this was the main motivator for going with separate packages or a separate suite package.

Why mix and match is important: There are a lot of v8 components we won't replace with converged for quite some time. DetailsList, ColorPicker, etc. Apps will need to use these v8 components along side converged, just like we need to use v0 alongside converged for quite some time.

Reply:
@Hotell on your graphic, both `FluentV8` and `NorthStar` need to exist for the foreseeable future. Though we do plan to fold converged components back into `@fluentui/react` once they are v1.0 replacing existing components (and updating FluentV8 components to use them, which would mean a major rev)

That being said doing a Major Rev of `@fluentui/react` right now is very difficult right now due to how partners consume it. So betting on being able to do a major rev of `@fluentui/react` in order for folks to adopt converged components is pretty risky. This was one of the motivators for separate packages and no dependencies on existing. 

We should, however, target folding back into `@fluentui/react` as that's where we have the reach, lots of partners and 3rd parties depend on this library (or on `office-ui-fabric-react` which upgrades to `@fluentui/react`).